### PR TITLE
treewide: add mainProgram attribute to lisp compilers

### PR DESCRIPTION
**Pull request by tani**
Added mainProgram attribute to the meta section of the following compilers:
- Clozure Common Lisp (ccl)
- CMU Common Lisp (cmucl)
- MKCL (mkcl)

This change helps in identifying the main executable for each compiler.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by uninsane** [comment]
changes look good. commit message should be changed before merging. our conventions are [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions), though the edge case here of changes small enough to not warrant distinct commits but which affect multiple packages is under-documented. in this case, we write it "treewide: <description of changes>", like "treewide: add `meta.mainProgram` to lisp compilers". [here's a recent example](https://github.com/NixOS/nixpkgs/pull/354024) of a similar commit.

File: pkgs/development/compilers/mkcl/default.nix
Comment (uninsane) [inline pkgs/development/compilers/mkcl/default.nix:60]: all else being equal, we tend to sort these attribute sets alphabetically. not a strict requirement but while you're editing the commit message consider moving this to be between `platforms` and `maintainers` & same for the other files.

**Comment by tani** [comment]
Thank you, @uninsane!

I've updated the commit message as suggested and sorted the attributes alphabetically for better readability and consistency. Please have a look and let me know if there’s anything else to address!

**Review by uninsane — APPROVED** [review]

**Review by wolfgangwalther — APPROVED** [review]

**Pull request merged**
